botorch/acquisition/analytic.py

```diff
@@ -1080,24 +1079,50 @@ def _get_noiseless_fantasy_model(
             "Only SingleTaskGP models with known observation noise "
             "are currently supported for fantasy-based NEI & LogNEI."
         )
+
     # initialize a copy of SingleTaskGP on the original training inputs
     # this makes SingleTaskGP a non-batch GP, so that the same hyperparameters
     # are used across all batches (by default, a GP with batched training data
     # uses independent hyperparameters for each batch).
+
+    # Could pass in the outcome_transform and input_transform here,
+    # however that would make them be applied in SingleTaskGP.__init__ which is
+    # unnecessary. So we will instead set them afterwards.
     fantasy_model = SingleTaskGP(
```

Comment:
This change would allow for supporting more model types. However, @saitcakmak  had a good question: Why do we need `_get_noiseless_fantasy_model` at all? Can we use the `fantasize` method on the model instead? I'm a bit afraid of the change I'm suggesting, since this instantiation logic won't be right for every model.

```suggestion
    fantasy_model = cls(model)(
```

Reply:
Yeah, it'd be a great simplification (& removal of duplicate logic) if we can simply use `model.fantasize(...)` rather than defining a custom `_get_noiseless_fantasy_model`.

Reply:
I think you mean to write `model.__class__(...)`; `cls(model)` is not a thing in Python.
But anyway, how about just not making that change because the model is currently assumed to be a `SingleTaskGP` anyway, and if we can find a way to use `fantasize` for a wider variety of models later, then just do that.